scripts/linters/pylint_extensions.py

```diff
@@ -2109,6 +2109,44 @@ def visit_call(self, node):
                     break
 
 
+class DisallowBlankLinesBelowFunctionDefinitionChecker(checkers.BaseChecker):
+    """Custom pylint checker to ensure there are no blank lines
+    below function definition.
+    """
+
+    _implements_ = interfaces.IAstroidChecker
+    name = 'disallow-blank-lines-below-function-definition'
+    priority = -1
+    msgs = {
+        'C0035': (
+            'Please remove the blank line(s) below the function definition.',
+            'remove-blank-lines-below-function-definiton.',
+            'Disallow blank lines below function definition.'
+        ),
+    }
+
+    def visit_functiondef(self, node):
+        """Visits every function and ensures there are no blank lines below
+        function definition.
+
+        Args:
+            node: astroid.scoped_nodes.Function. Node for a function or method
+                definition in the AST.
+        """
+        line_number = node.fromlineno
+        while True:
+            line = linecache.getline(node.root().file, line_number).strip()
+            if line.startswith((b'\'', b'"')):
```

Comment:
I think you are still assuming all functions to have a docstring, right? But consider your own test methods you wrote in this PR - even they will fail the lint check using this logic, but they should pass.

I have a simple logic for this checker - starting from first line of the function, the first line that you find ending with ":" (colon) is where the function ends. If you find a blank line after that you raise an error. Taking this line of thought, try to frame the solution that covers all the cases.

Reply:
Yes sorry I forgot to change the code. I have implemented the class  by your logic now.  Its corrected now.